### ISSUE TITLE
Access video dimensions for scaling purposes

### ISSUE THREAD

**Issue by Sir-hennihau** [TheWidlarzGroup/react-native-video#746]
Hi!
I want to set the video player size related to the video dimensions.

According to: https://github.com/react-native-community/react-native-video/issues/203 : 

> The dimensions are posted in onLoad event body.

I don't know how to do that and to get the values from there.

My code shall look like: 
```
<Video

   style={{
      width: this.state.screenWidth,
      height: this.state.heightScaled }}

    source={{ uri: this.props.uriProp }}
    resizeMode='cover'
    paused={this.state.videoPaused}
    onLoad={() => {
       this.setState({
            videoPaused: false,
        });
      }}
/>
```
"heightScaled" shall be calculated using the video dimensions videoWidth and videoHeight. The result shall be a video using full screen width and a aspect ratio containg height.

**tldr**: How can I access the video dimensions in onLoad() ?

Thanks in advance.
Sir hennihau

**Comment by GertjanReynaert** [comment]
@Sir-hennihau This should do the trick:

```jsx
<Video
  style={{
    width: this.state.screenWidth,
    height: this.state.heightScaled
  }}
  source={{ uri: this.props.uriProp }}
  resizeMode='cover'
  paused={this.state.videoPaused}
  onLoad={response => {
    const { width, height } = response.naturalSize;
    const heightScaled = height * (this.state.screenWidth / width);

    this.setState({
      heightScaled,
      videoPaused: false,
    });
  }}
/>
```

**Comment by cmmartin** [comment]
This can be closed

**Comment by borie88** [comment]
I recently had an issue where onLoad wasn't returning a naturalSize object. From the documentation, I found that onReadyForDisplay returns naturalSize. Switched to that and everything works fine

**Comment by ram95krishh** [comment]
@borie88 In react native video 4.4.4, it returns an object { target: 4567 } with some number in it. I believe it is supposed to be some callback to the native video player. I don't find natural size is neither onLoad nor onReadyForDisplay. :/

![image](https://user-images.githubusercontent.com/45874777/64967951-fe19cb80-d8be-11e9-931b-f959b65d8827.png)

**Comment by ericp3reira** [comment]
I get `naturalSize` on `onLoad`, but all its attributes are `undefined`, even if I explicitly set `height` and `width`.

**Comment by hanhuizhu** [comment]
@ericp3reira ,i have the same problem, do you resolve it ?

**Comment by eaazumah** [comment]
This work for me
`const [videoStyle, setVideoStyle] = useState({
    width: width,
    height: width,
  });
const onVideoLoad = (response: any) => {
    const size = response.naturalSize;
    if (size?.width && size?.height) {
      if (size?.width < size?.height) {
        const aspectRatio = size?.height / size?.width;
        const newHeight =
          width * aspectRatio < height * 0.55
            ? width * aspectRatio
            : height * 0.55;
        if (newHeight) {
          setVideoStyle({
            width,
            height: newHeight,
          });
        }
        return;
      }
      const aspectRatio = size?.width / size?.height;
      const newHeight =
        width * aspectRatio < height * 0.55
          ? width * aspectRatio
          : height * 0.55;
      if (newHeight) {
        setVideoStyle({
          width,
          height: newHeight,
        });
      }
    }
  };

return (
    <VideoPlayer
      resizeMode={'contain'}
      muted={false}
      repeat={true}
      tapAnywhereToPause={true}
      playInBackground={false}
      playWhenInactive={false}
      ignoreSilentSwitch={'ignore'}
      progressUpdateInterval={250.0}
      disableSeekbar={true}
      disableFullscreen={true}
      disablePlayPause={true}
      disableTimer={true}
      disableVolume={true}
      disableBack={true}
      onLoad={onVideoLoad}
      source={{
        uri: videoURI ? videoURI : url,
        type: 'mp4',
      }}
      style={videoStyle}
    />
  );
`

**Comment by kartikeyvaish** [comment]
@eaazumah's comment, but with indentation

```javascript
const [videoStyle, setVideoStyle] = useState({
  width: width,
  height: width,
});

const onVideoLoad = (response: any) => {
  const size = response.naturalSize;
  if (size?.width && size?.height) {
    if (size?.width < size?.height) {
      const aspectRatio = size?.height / size?.width;
      const newHeight =
        width * aspectRatio < height * 0.55
          ? width * aspectRatio
          : height * 0.55;
      if (newHeight) {
        setVideoStyle({
          width,
          height: newHeight,
        });
      }
      return;
    }

    const aspectRatio = size?.width / size?.height;
    
    const newHeight =
      width * aspectRatio < height * 0.55 ? width * aspectRatio : height * 0.55;
    if (newHeight) {
      setVideoStyle({
        width,
        height: newHeight,
      });
    }
  }
};

return (
  <VideoPlayer
    resizeMode={'contain'}
    muted={false}
    repeat={true}
    tapAnywhereToPause={true}
    playInBackground={false}
    playWhenInactive={false}
    ignoreSilentSwitch={'ignore'}
    progressUpdateInterval={250.0}
    disableSeekbar={true}
    disableFullscreen={true}
    disablePlayPause={true}
    disableTimer={true}
    disableVolume={true}
    disableBack={true}
    onLoad={onVideoLoad}
    source={{
      uri: videoURI ? videoURI : url,
      type: 'mp4',
    }}
    style={videoStyle}
  />
);
```
